Cargo.toml

```diff
@@ -59,6 +59,9 @@ regex = ["dep:regex-automata"]
 # Enable serde serialization support
 serde = ["dep:serde"]
 
+# Enable dependencies only needed for generation of documentation on docs.rs
+docsrs = ["dep:vergen"]
```

Comment:
Have you checked that this actually gets enabled with generating docsrs docs?

Reply:
I haven't tested it with the live docsrs, since I'm not sure how one would go about doing so.

My understanding is that the `docsrs` feature gets enabled by
https://github.com/zesterer/chumsky/blob/af4e2b12cf1447e0e321af80040975254a4aa6a4/Cargo.toml#L67-L69

Reply:
I don't think this would enable such a feature. The cfg flag just allows you to do `#[cfg(docsrs)]` (which would probably be sufficient for your purposes, fwiw, instead of adding a cargo feature).

Reply:
Okay, I tried a few things in that direction, but haven't quite managed to get it to work as expected.

So I first tested the code with `RUSTFLAGS="--cfg docsrs" cargo doc --open`. That only seems to pass the info to the Cargo.toml part. It doesn't pass the info to the `lib.rs` file, since  `#[cfg(docsrs)]` in `lib.rs` had no effect.

Then I tested it with `cargo rustdoc --open -- --cfg docsrs`. This gets the lib.rs file to cooperate, but it fails to do any of the vergen stuff.

I could probably find a variation where I get it to work, but it might almost be easier to make the `docsrs` stuff a feature.

Reply:
Have you tried the command at the top of `guide.rs`?

```
RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --all-features
```

Reply:
I missed that command, thank you.

So I tested it just now:
Conditionally compiling vergen does not seem to be possible with
```
[target.'cfg(docsrs)'.build-dependencies]
vergen = { version = "=8.1.1", features = ["git", "gitoxide"] }
```

Conditionally compiling it with `--all-features` and then `docsrs = ["dep:vergen"]` does seem to work.

Reply:
Ah, that's unfortunate :(

Reply:
Hmm, not sure I understood this message. So this does work?

Reply:
Yep, this pull request should work.